### PR TITLE
Document the staging and production release of the application

### PR DESCRIPTION
**Pull request by rowlando**
## What does this pull request do?

Documents the release process for this application for both non-production and production releases.

🔎 Nicer link to the new section: https://github.com/ministryofjustice/fala/blob/instructions-for-release/README.md#releasing

fala-release-process

## Any other changes that would benefit highlighting?

We will switch the default branch on this repository from `develop` to `master` once this is merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,3 +49,33 @@ gulp serve
...
+
+1. Wait for [the Docker build to complete on CircleCI](https://circleci.com/gh/ministryofjustice/fala) for the feature branch.
+1. Copy the `feature_branch.<sha>` reference from the `build` job's "Push Docker image" step. Eg:
+    ```
+    Pushing tag for rev [3e1a4ef9ecf6] on {https://registry.service.dsd.io/v1/repositories/fala/tags/github-config.843dcc7}
+    ```
+1. [Deploy `feature_branch.<sha>`](https://ci.service.dsd.io/job/DEPLOY-fala/build?delay=0sec).
+    * `ENVIRONMENT` is the target environment, select "staging".
+    * `DEPLOY_BRANCH` is the branch that needs to be released (`github-config` in the above example).
+    * `VERSION` is either `latest` for the last successful build on that branch, or a specific 7-character prefix of the Git SHA (`843dcc7` in the above example).
```
Comment (sldblog) [inline README.md]: For this repository, these variables seem to be different:

- `version` seems to be the concatenated `branch.sha` format
- `deploy_branch` seems to be correlating to the deploy repo's branch name.

screen shot 2018-08-30 at 15 38 14

File: README.md [hunk 2]
```diff
@@ -49,3 +49,33 @@ gulp serve
...
+1. [Deploy `feature_branch.<sha>`](https://ci.service.dsd.io/job/DEPLOY-fala/build?delay=0sec).
+    * `ENVIRONMENT` is the target environment, select "staging".
+    * `DEPLOY_BRANCH` is the [deploy repo's](https://github.com/ministryofjustice/fala-deploy) default branch name, usually `master`.
+    * `VERSION` is the branch that needs to be released plus a specific 7-character prefix of the Git SHA. (`github-config.843dcc7` for the above example).
+
+
+### Releasing to production
+
+1. Please make sure you tested on a non-production environment before merging.
+1. Merge your feature branch pull request to `master`.
```
Comment (sldblog) [inline README.md]: 🙋‍♂️ This repository does not use `master` as the default branch. Should we switch?

Comment (rowlando) [reply README.md]: Yeap, that's why I wrote:

>We will switch the default branch on this repository from develop to master once this is merged.

in the pull request description.

We should do a release as soon as we merge develop into master, then set `master` as default branch.

Comment (sldblog) [reply README.md]: Right, I missed that, sorry :(

**Pull request merged**